storage/postgres/postgres_adapter.go

```diff
@@ -0,0 +1,198 @@
+package postgres
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	_ "github.com/lib/pq" // Imported as a side-effect to register drivers with database/sql package
+
+	"BNR-Blog-Dockertest/storage"
+)
+
+const (
+	sslModeKey  = "sslmode"
+	passwordKey = "password"
+	hostKey     = "host"
+	portKey     = "port"
+	userKey     = "user"
+	dbNameKey   = "dbname"
+)
+
+// PgAdapter represents the postgres storage adapter
+type PgAdapter struct {
+	tableName string
+	conn      *sql.DB
+}
+
+// createTable creates a new table based on constant values
+func (a PgAdapter) createTable(tableName string) error {
+	sqlStatement := fmt.Sprintf("CREATE TABLE %s (id SERIAL PRIMARY KEY,number TEXT);", tableName)
+
+	_, err := a.conn.Exec(sqlStatement)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// pgOptions describes connection options
+type pgOptions struct {
+	passWord  string
+	tableName string
+	sslMode   string
+	host      string
+	port      string
+	userName  string
+	dbName    string
+}
+
+// PgOptionFunc describes functions which add optional connection variables to Postgres
+type PgOptionFunc func(options *pgOptions)
```

Comment:
I happened upon this PR because I'm interested in the topic. I'm not sure why this type is exported, but the function parameter isn't. Should this type be unexported as well?

Reply:
Ah good comment. This is currently exported because other layers access this option within the overall app. But for the purposes of this PR it can probably be unexported.

Reply:
@drognisep I updated the options struct to be exported with a reason for why this is done. Let me know if this is clear to you.